derive_builder/src/options/darling_opts.rs

```diff
@@ -0,0 +1,486 @@
+            name: Ident::from("build"),
+            validate: None,
+        }
+    }
+}
+
+/// Contents of the `field` meta in `builder` attributes.
+#[derive(Debug, Clone, Default, FromMeta)]
+#[darling(default)]
+pub struct FieldMeta {
+    public: Flag,
+    private: Flag,
+}
+
+impl FlagVisibility for FieldMeta {
+    fn public(&self) -> &Flag {
+        &self.public
+    }
+
+    fn private(&self) -> &Flag {
+        &self.private
+    }
+}
+
+#[derive(Debug, Clone, Default, FromMeta)]
+#[darling(default)]
+pub struct StructLevelSetter {
+    prefix: Option<Ident>,
+    into: Option<bool>,
+    skip: Option<bool>,
+}
+
+impl StructLevelSetter {
+    /// Check if setters are explicitly enabled or disabled at
+    /// the struct level.
+    pub fn enabled(&self) -> Option<bool> {
+        self.skip.map(ops::Not::not)
+    }
+}
+
+#[derive(Debug, Clone, Default, FromMeta)]
+#[darling(default)]
+pub struct FieldLevelSetter {
+    prefix: Option<Ident>,
+    name: Option<Ident>,
+    into: Option<bool>,
+    skip: Option<bool>,
+}
+
+impl FieldLevelSetter {
+    /// Get whether or not this field-level setter indicates a setter should
+    /// be emitted. The setter shorthand rules are that the presence of a
+    /// `setter` with _any_ properties set forces the setter to be emitted.
+    pub fn enabled(&self) -> Option<bool> {
+        if self.skip.is_some() {
+            return self.skip.map(ops::Not::not);
+        }
+
+        if self.prefix.is_some() || self.name.is_some() || self.into.is_some() {
+            return Some(true);
+        }
+
+        None
+    }
+}
+
+#[derive(Debug, Clone)]
+enum FieldSetterMeta {
+    Shorthand,
+    Longhand(FieldLevelSetter),
+}
+
+impl From<FieldSetterMeta> for FieldLevelSetter {
+    fn from(v: FieldSetterMeta) -> Self {
+        match v {
+            FieldSetterMeta::Shorthand => FieldLevelSetter {
+                skip: Some(false),
+                ..Default::default()
+            },
+            FieldSetterMeta::Longhand(val) => val,
+        }
+    }
+}
+
+impl FromMeta for FieldSetterMeta {
+    fn from_word() -> darling::Result<Self> {
+        Ok(FieldSetterMeta::Shorthand)
+    }
+
+    fn from_nested_meta(value: &syn::NestedMeta) -> darling::Result<Self> {
+        Ok(FieldSetterMeta::Longhand(
+            FieldLevelSetter::from_nested_meta(value)?,
+        ))
+    }
+
+    fn from_list(value: &[syn::NestedMeta]) -> darling::Result<Self> {
+        Ok(FieldSetterMeta::Longhand(FieldLevelSetter::from_list(
+            value,
+        )?))
+    }
+}
+
+#[derive(Debug, Clone, FromField)]
+#[darling(attributes(builder), forward_attrs(doc, cfg, allow))]
+pub struct Field {
+    ident: Option<Ident>,
+    attrs: Vec<Attribute>,
+    vis: syn::Visibility,
+    ty: syn::Type,
+    #[darling(default)]
+    pattern: Option<BuilderPattern>,
+    #[darling(default)]
+    public: Flag,
+    #[darling(default)]
+    private: Flag,
+    #[darling(default, map = "FieldSetterMeta::into")]
```

Comment:
This needs an explanatory comment. We're using `map` to parse into a completely different type than the field declares, and then using `into` to convert from there into `FieldLevelSetter`.